plugins/auth-react/src/hooks/useCookieAuthRefresh/useCookieAuthRefresh.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useCallback } from 'react';
+import {
+  discoveryApiRef,
+  fetchApiRef,
+  storageApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import { useAsync, useMountEffect } from '@react-hookz/web';
+import { ResponseError } from '@backstage/errors';
+
+/**
+ * @public
+ * A hook that will refresh the cookie when it is about to expire.
+ * @param options - Options for configuring the refresh cookie endpoint
+ */
+export function useCookieAuthRefresh(options: {
+  // The plugin id used for discovering the API origin
+  pluginId: string;
+  // The path used for calling the refresh cookie endpoint, default to '/cookie'
+  path?: string;
+}):
+  | { status: 'loading' }
+  | { status: 'error'; error: Error; retry: () => void }
+  | { status: 'success'; data: { expiresAt: string } } {
+  const { pluginId, path = '/cookie' } = options ?? {};
+  const fetchApi = useApi(fetchApiRef);
+  const storageApi = useApi(storageApiRef);
+  const discoveryApi = useApi(discoveryApiRef);
+
+  const store = storageApi.forBucket(`${pluginId}-auth-cookie-storage`);
```

Comment:
Oh we actually can't use the storage API for this, sorry. It is often connected to a global store (through the user-settings-backend) so the state would be shared across all devices. But cookies are by definition constrained to the current browser instance. So every device needs to explicitly only handle its own state locally, otherwise it'll mess up other devices connected to the same store. The only exception was the cross-tab sharing of course - but that needed to use message passing for pings.

Reply:
Good catch, thanks for pointing it out 🙏🏻 
I updated the hook to use Broadcast Channel again.